src/common/transformations/src/transformations/common_optimizations/gelu_fusion.cpp

```diff
@@ -267,69 +225,79 @@ ov::pass::GeluFusionWithTanhNoPower::GeluFusionWithTanhNoPower() {
-                pattern_to_output.at(add1).get_node_shared_ptr(),
-                pattern_to_output.at(mul3).get_node_shared_ptr(),
-                pattern_to_output.at(mul4).get_node_shared_ptr(),
-                pattern_to_output.at(tanh).get_node_shared_ptr(),
-                pattern_to_output.at(add2).get_node_shared_ptr(),
-                pattern_to_output.at(mul5).get_node_shared_ptr(),
-                pattern_to_output.at(mul6).get_node_shared_ptr(),
-            },
-            gelu);
-        ov::replace_node(m.get_match_root(), gelu);
-        return true;
+    ov::matcher_pass_callback callback = [=](ov::pass::pattern::Matcher& m) {
+        return gelu_replacer(m, input, op::GeluApproximationMode::TANH);
+    };
+
+    auto m = std::make_shared<ov::pass::pattern::Matcher>(mul6, matcher_name);
+    register_matcher(m, callback);
+}
+
+ov::pass::GeluFusionWithTanhNoPower2::GeluFusionWithTanhNoPower2() {
+    MATCHER_SCOPE(GeluFusionWithTanhNoPower2);
+    // Replaces a sub-graph with a Gelu (ov::op::v0::Tanh) op
+    // x * (0.5 * (1 + tanh([sqrt(2 / pi)] * [x + 0.044715 * x * x * x])))
+    auto input = pass::pattern::any_input();
+
+    auto mul_0 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({input, input});
+    auto mul_1 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({input, mul_0});
+
+    auto mul_2_constant = ov::pass::pattern::wrap_type<ov::op::v0::Constant>(check_value(0.044715f, 0.001f));
+    auto mul_2 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({mul_1, mul_2_constant});
+
+    auto add_0 = ov::pass::pattern::wrap_type<ov::op::v1::Add>({input, mul_2});
+
+    auto mul_3_constant = ov::pass::pattern::wrap_type<ov::op::v0::Constant>(check_value(SQRT_2_PI, 0.01f));
+    auto mul_3 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({add_0, mul_3_constant});
+
+    auto tanh = ov::pass::pattern::wrap_type<ov::op::v0::Tanh>({mul_3});
+
+    auto add_1_constant = ov::pass::pattern::wrap_type<ov::op::v0::Constant>(check_value(1.0f));
+    auto add_1 = ov::pass::pattern::wrap_type<ov::op::v1::Add>({tanh, add_1_constant});
+
+    auto mul_4_constant = ov::pass::pattern::wrap_type<ov::op::v0::Constant>(check_value(0.5f));
+
+    // x * (0.5 * (1 + tanh))
+    auto mul_4_1 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({add_1, mul_4_constant});
+    auto mul_5_1 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({input, mul_4_1});
+
+    // (x * 0.5) * (1 + tanh)
+    auto mul_4_2 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({input, mul_4_constant});
+    auto mul_5_2 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({add_1, mul_4_2});
+
+    // 0.5 * ( x * (1 + tanh))
+    auto mul_4_3 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({add_1, input});
+    auto mul_5_3 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({mul_4_3, mul_4_constant});
+
+    auto mul_5 = std::make_shared<ov::pass::pattern::op::Or>(OutputVector{mul_5_1, mul_5_2, mul_5_3});
```

Comment:
could you please cover all the cases with unit tests?
As I can see, there are 3 options here, but only one tested

Reply:
Applied, thanks!